x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/ranges.test.tsx

```diff
@@ -744,6 +744,55 @@ describe('ranges', () => {
           /^Bytes format:/
         );
       });
+
+      it('should not reset formatters when switching between custom ranges and auto histogram', () => {
+        const setStateSpy = jest.fn();
+
+        // Add an extra range
+        (state.layers.first.columns.col1 as RangeIndexPatternColumn).params.ranges.push({
+          from: DEFAULT_INTERVAL,
+          to: 2 * DEFAULT_INTERVAL,
+          label: '',
+        });
+
+        state.indexPatterns['1'].fieldFormatMap = {
+          MyField: { id: 'custom', params: {} },
+        };
+
+        // now set a format on the range operation
+        (state.layers.first.columns.col1 as RangeIndexPatternColumn).params.format = {
+          id: 'custom',
+          params: { decimals: 0 },
```

Comment:
It seems like the only difference between the "default" format of `MyField` and the overwrite on the column are the params. Your test doesn't really validate these params are kept - I commented out lines 763 to 766 and the tests still pass. I suggest not setting the same formatter in the field format map of the index pattern (line 759)

Reply:
Not sure if I understood that completely. Should I switch it to test for params on the same line(759)?

Reply:
No, I meant setting it to the number formatter there so you can actually check whether the format is persisted.

But I just realized your test is not testing the right thing anyway. `lns-customBucketContainer-remove` is removing one out of multiple ranges, but that's not what should be tested here - this is about switching between custom ranges in general and automatic ranges controlled by granularity.

A proper unit test should click the "Remove custom ranges" button, then check whether the state update issues by this still includes the custom format.

Let me know when you have problems adding such a test.

Reply:
> No, I meant setting it to the number formatter there so you can actually check whether the format is persisted.
> 
> But I just realized your test is not testing the right thing anyway. `lns-customBucketContainer-remove` is removing one out of multiple ranges, but that's not what should be tested here - this is about switching between custom ranges in general and automatic ranges controlled by granularity.
> 
> A proper unit test should click the "Remove custom ranges" button, then check whether the state update issued by this still includes the custom format.
> 
> Let me know when you have problems adding such a test. I can do it and add it to your PR

So, I have been trying to
```
act(() => {
          instance
            .find(".euiLink")
            .first()
            .prop('onClick')!({} as ReactMouseEvent);

          expect(setStateSpy).toHaveBeenCalledWith({
            ...state,
            layers: {
              first: {
                ...state.layers.first,
                columns: {
                  ...state.layers.first.columns,
                  col1: {
                    ...state.layers.first.columns.col1,
                    params: {
                      ...state.layers.first.columns.col1.params,
                      format: {
                        id: "custom",
                        params: { decimals: 3 }
                      }
                    },
                  },
                },
              },
            },
          });
        })
```
But it seems that ```Number of calls: 2``` and I'm not sure how to handle that